modules/openthread/Kconfig

```diff
@@ -3,11 +3,327 @@
 
 config OPENTHREAD
 	bool "OpenThread Support"
+	imply FLASH
+	imply FLASH_MAP
+	imply MPU_ALLOW_FLASH_WRITE
+
+	select SETTINGS if FLASH
+	select OPENTHREAD_SETTINGS_RAM if !FLASH
+	select CPP
+	select REBOOT
+	select ENTROPY_GENERATOR
+
+	select NRF_802154_RADIO_DRIVER if HAS_HW_NRF_RADIO_IEEE802154 && !NET_L2_OPENTHREAD
+	select NRF_802154_SER_HOST if !HAS_HW_NRF_RADIO_IEEE802154 && !NET_L2_OPENTHREAD
+	select NRF_802154_RADIO_CONFIG if !NET_L2_OPENTHREAD
 	help
 	  This option enables the OpenThread library
 
 if OPENTHREAD
 
+choice OPENTHREAD_IMPLEMENTATION
+	prompt "OpenThread origin selection"
+	help
+	  Select OpenThread stack to use for build. Custom OpenThread implementations
+	  can be added to the application Kconfig.
+
+config OPENTHREAD_SOURCES
+	bool "OpenThread from sources"
+	help
+	  Build Zephyr's OpenThread port from sources.
+
+endchoice
+
+config OPENTHREAD_MANUAL_START
+	bool "Start OpenThread stack manually"
+	help
+	  If enabled, OpenThread stack will have to be configured and
+	  started manually, with respective API calls or CLI/NCP commands.
+	  Otherwise, OpenThread will configure the network parameters and try to
+	  join the Thread network automatically during initialization (using
+	  credentials stored in persistent storage, obtained during
+	  commissioning or pre-commissioned with other Kconfig options,
+	  depending on configuration used).
+
+menu "Logging"
+
+menuconfig OPENTHREAD_DEBUG
+	bool "OpenThread stack logging support"
+	help
+	  This option enables logging support for OpenThread.
+
+choice OPENTHREAD_LOG_LEVEL_CHOICE
+	prompt "OpenThread stack log level"
+	depends on OPENTHREAD_DEBUG
+	help
+	  This option selects log level for OpenThread stack.
+
+config OPENTHREAD_LOG_LEVEL_CRIT
+	bool "Critical"
+config OPENTHREAD_LOG_LEVEL_WARN
+	bool "Warning"
+config OPENTHREAD_LOG_LEVEL_NOTE
+	bool "Notice"
+config OPENTHREAD_LOG_LEVEL_INFO
+	bool "Informational"
+config OPENTHREAD_LOG_LEVEL_DEBG
+	bool "Debug"
+endchoice # OPENTHREAD_LOG_LEVEL_CHOICE
+
+config OPENTHREAD_LOG_LEVEL
+	int
+	default 1 if OPENTHREAD_LOG_LEVEL_CRIT
+	default 2 if OPENTHREAD_LOG_LEVEL_WARN
+	default 3 if OPENTHREAD_LOG_LEVEL_NOTE
+	default 4 if OPENTHREAD_LOG_LEVEL_INFO
+	default 5 if OPENTHREAD_LOG_LEVEL_DEBG
+	default 0
+	help
+	  Log level for OpenThread stack.
+
+config OPENTHREAD_PLATFORM_LOG_LEVEL
+	int
+	default 1 if OPENTHREAD_LOG_LEVEL_CRIT
+	default 2 if OPENTHREAD_LOG_LEVEL_WARN
+	default 3 if OPENTHREAD_LOG_LEVEL_NOTE || OPENTHREAD_LOG_LEVEL_INFO
```

Comment:
this should all be using e.g.
```
module = MCUMGR_GRP_OS
module-str = mcumgr_grp_os
source "subsys/logging/Kconfig.template.log_config"
```

Reply:
It is copied from https://github.com/zephyrproject-rtos/zephyr/blob/54b826336bae437f851fcba332481e24a7e1532d/modules/openthread/Kconfig#L73-L104

Reply:
and from 2 weeks ago... @ArekBalysNordic please fix this upstream, this is not how logging Kconfigs are done

Reply:
But this config is not meant to control Zephyr logger in the first place but OpenThread logger, so Zephyr rules don't apply here. `OPENTHREAD_PLATFORM_LOG_LEVEL` then only maps OpenThread logging level to Zephyr logging level.

Reply:
OK on that, the style is still in breach of zephyr guidelines so that needs to be fixed https://docs.zephyrproject.org/latest/contribute/style/kconfig.html